bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/connection/PhilipsAirAPIConnection.java

```diff
@@ -0,0 +1,242 @@
+                throw new PhilipsAirAPIException(
+                        "Cooldown period is active, waiting Philips Air Purifier device responded with status code");
+            }
+
+            Request request = httpClient.newRequest(url).method(method);
+            if (method == PUT && StringUtils.isNotEmpty(content)) {
+                request.content(new StringContentProvider(content));
+            }
+
+            ContentResponse contentResponse = request.timeout(config.getRefreshInterval(), TimeUnit.SECONDS).send();
+            int httpStatus = contentResponse.getStatus();
+            String finalcontent = contentResponse.getContentAsString();
+            if (decode) {
+                try {
+                    finalcontent = this.cipher.decrypt(finalcontent);
+                } catch (BadPaddingException bexp) {
+                    // retry for once with a new key
+                    config.setKey("");
+                    initCipher();
+                    finalcontent = this.cipher.decrypt(finalcontent);
+                }
+            }
+
+            logger.debug("Philips Air Purifier device response: status = {}, content = '{}'", httpStatus, finalcontent);
+            switch (httpStatus) {
+                case OK_200:
+                    return finalcontent;
+                case BAD_REQUEST_400:
+                case UNAUTHORIZED_401:
+                case NOT_FOUND_404:
+                    logger.debug("Philips Air Purifier device responded with status code {}", httpStatus);
+                    throw new PhilipsAirAPIException(String.format("Error with status %d", httpStatus));
+                case TOO_MANY_REQUESTS_429:
+                    cooldownTimer = System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(5);
+                default:
+                    logger.debug("Philips Air Purifier device responded with status code {}", httpStatus);
+                    throw new PhilipsAirAPIException(String.format("Error with status %d", httpStatus));
+            }
+        } catch (ExecutionException e) {
+            String errorMessage = e.getLocalizedMessage();
+            logger.trace("Exception occurred during execution: {}", errorMessage, e);
+            throw new PhilipsAirAPIException(errorMessage, e.getCause());
+        } catch (InterruptedException | TimeoutException e) {
+            logger.debug("Exception occurred during execution: {}", e.getLocalizedMessage(), e);
+            throw new PhilipsAirAPIException(e.getLocalizedMessage(), e.getCause());
+        } catch (Exception e) {
+            logger.error("Unexpected exception occurred during execution: {}", e.getLocalizedMessage(), e);
+            throw new PhilipsAirAPIException(e.getLocalizedMessage(), e.getCause());
+        }
+    }
+
+    public String exchangeKeys() throws GeneralSecurityException, InterruptedException, TimeoutException,
+            ExecutionException, InvalidAlgorithmParameterException {

```

Comment:
It is unreasonable to expect that callers of this method could, or even should, handle all these different types of exceptions. You can simplify it by wrapping certain exceptions in a `PhilipsAirAPIException` and then throwing that instead. The only thing you shouldn't wrap is the `InterruptedException` since you should never try to wrap them.
```suggestion
    public String exchangeKeys() throws PhilipsAirAPIException, InterruptedException {
```